examples/webgl_custom_clipping.html

```diff
@@ -0,0 +1,399 @@
+					mesh.matrixAutoUpdate = false;
+
+					volumeVisualization.add( mesh );
+
+				}
+
+				scene.add( volumeVisualization );
+
+
+				var ground = new THREE.Mesh( planeGeometry,
+						new THREE.MeshPhongMaterial( {
+							color: 0xa0adaf, shininess: 150 } ) );
+				ground.rotation.x = - Math.PI / 2;
+				ground.scale.multiplyScalar( 3 );
+				ground.receiveShadow = true;
+				scene.add( ground );
+
+				// Renderer
+
+				var container = document.body;
+
+				renderer = new THREE.WebGLRenderer();
+				renderer.shadowMap.enabled = true;
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				window.addEventListener( 'resize', onWindowResize, false );
+				container.appendChild( renderer.domElement );
+				// Clipping setup:
+				globalClippingPlanes = createPlanes( GlobalClippingPlanes.length );
+				renderer.clippingPlanes = Empty;
+				renderer.localClippingEnabled = true;
+
+				// Stats
+
+				stats = new Stats();
+				container.appendChild( stats.domElement );
+
+				// Controls
+
+				var controls = new THREE.OrbitControls( camera, renderer.domElement );
+				controls.target.set( 0, 1, 0 );
+				controls.update();
+
+				// GUI
+
+				var gui = new dat.GUI(),
+					folder = gui.addFolder( "Local Clipping" ),
+					props = {
+						get 'Enabled'() { return renderer.localClippingEnabled; },
+						set 'Enabled'( v ) {
+								renderer.localClippingEnabled = v;
+								if ( ! v ) volumeVisualization.visible = false; },
+
+						get 'Shadows'() { return clipMaterial.clipShadows; },
+						set 'Shadows'( v ) { clipMaterial.clipShadows = v; },
+
+						get 'Visualize'() { return volumeVisualization.visible; },
+						set 'Visualize'( v ) {
+								if ( renderer.localClippingEnabled )
+									volumeVisualization.visible = v; }
+					};
+
+				folder.add( props, 'Enabled' );
+				folder.add( props, 'Shadows' );
+				folder.add( props, 'Visualize' ).listen();
+
+				gui.addFolder( "Global Clipping" ).
+						add( {
+							get 'Enabled'() { return renderer.clippingPlanes !== Empty; },
+							set 'Enabled'( v  ) { renderer.clippingPlanes = v ?
+									globalClippingPlanes : Empty; }
+						}, "Enabled" );
+
+
+				// Start
+
+				startTime = Date.now();
+
+			}
+
+			function onWindowResize() {
+
+				camera.aspect = window.innerWidth / window.innerHeight;
+				camera.updateProjectionMatrix();
+
+				renderer.setSize( window.innerWidth, window.innerHeight );
+
+			}
+
+			var transform = new THREE.Matrix4(),
+				tmpMatrix = new THREE.Matrix4();
+
+			function animate() {
+
+				var currentTime = Date.now(),
+					time = ( currentTime - startTime ) / 1000;
+
+				requestAnimationFrame( animate );
+
+				transform.makeTranslation( 0, 1, 0 ).
+						multiply( tmpMatrix.makeRotationX( time * 0.5 ) ).
+						multiply( tmpMatrix.makeRotationY( time * 0.2 ) );
+
+				object.matrix.copy( transform );
+				object.updateMatrixWorld( true );
```

Comment:
@WestLangley 

Ha! Yes, I thought that it might be a didactic mistake.

> Just set object.rotation.x/y/z.

OK. In this case I can use the object to compose the transform and then copy `.matrix` for the transform of the clipping volume.

However, I have to set the matrix for visualizing the planes (it'd be ill-advised to decompose it just for this purpose, I guess):

https://github.com/tschw/three.js/blob/ClippingPlanes/examples/webgl_custom_clipping.html#L256
https://github.com/tschw/three.js/blob/ClippingPlanes/examples/webgl_custom_clipping.html#L376-L377

What's the most advisable way to do so? I've actually been looking whether there is an `Object3D.setMatrix` routine... 

```javascript
object.matrix.makeIdentity();
object.applyMatrix( matrix );
```

Actually, since [this utility function](https://github.com/tschw/three.js/blob/ClippingPlanes/examples/webgl_custom_clipping.html#L92-L129) produces a world matrix, I might want to add another:

```javascript
function setObjectWorldMatrix( object, matrix ) {
    // cancel inherited world transform
    object.matrix.getInverse( object.parent.matrixWorld );
    // apply sought orientation
    object.applyMatrix( matrix );
}
```
The example does not need it, but this way it embeds more robust utility code for the user to copy over - and I don't have to touch `.matrixAutoUpdate`.

You like?